feature-libs/quote/components/details/comment/quote-details-comment.component.ts

```diff
@@ -79,27 +102,77 @@ export class QuoteDetailsCommentComponent {
         map((quote) => quote.isEditable)
       ),
       dateFormat: 'MMMM d, yyyy h:mm aa',
+      defaultItemId: ALL_PRODUCTS_ID,
+      itemList$: this.prepareItemList(),
     };
   }
 
+  private prepareItemList(): Observable<Item[]> | undefined {
+    let allProducts: string = 'quote.comments.allProducts';
+    this.translationService
+      .translate(allProducts)
+      .pipe(take(1))
+      .subscribe((text) => (allProducts = text));
+
+    return this.quoteDetails$.pipe(
+      map((quote) => {
+        const itemList: Item[] = [{ id: ALL_PRODUCTS_ID, name: allProducts }];
+        quote.entries?.forEach((entry) => {
+          const item = this.convertToItem(entry);
+          if (item) {
+            itemList.push(item);
+          }
+        });
+        return itemList;
+      })
+    );
+  }
+
+  private convertToItem(entry: OrderEntry | undefined): Item | undefined {
+    if (entry?.entryNumber !== undefined) {
```

Comment:
Do not return undefined. We always expect an entryNumber, so throw an error if it is not provided